newIDE/app/src/EventsFunctionsList/FunctionTreeViewItemContent.js

```diff
@@ -0,0 +1,441 @@
+      case gd.EventsFunction.Expression:
+      case gd.EventsFunction.ExpressionAndCondition:
+        return 'res/functions/expression.svg';
+    }
+  }
+
+  getDataset(): ?HTMLDataset {
+    return null;
+  }
+
+  onSelect(): void {
+    this.props.onSelectEventsFunction(
+      this.eventFunction,
+      this.props.eventsBasedBehavior,
+      this.props.eventsBasedObject
+    );
+  }
+
+  rename(newName: string): void {
+    if (this.eventFunction.getName() === newName) return;
+
+    this.props.onRenameEventsFunction(this.eventFunction, newName, doRename => {
+      if (!doRename) return;
+      this._onEventsFunctionModified();
+    });
+  }
+
+  edit(): void {}
+
+  buildMenuTemplate(i18n: I18nType, index: number) {
+    const eventsFunction = this.eventFunction;
+    return [
+      {
+        label: i18n._(t`Rename`),
+        click: () => this.props.editName(this.getId()),
+        enabled: this.props.canRename(eventsFunction),
+      },
+      {
+        label: eventsFunction.isPrivate()
+          ? i18n._(t`Make public`)
+          : i18n._(t`Make private`),
+        click: () => this._togglePrivate(eventsFunction),
+      },
+      {
+        label: eventsFunction.isAsync()
+          ? i18n._(t`Make synchronous`)
+          : i18n._(t`Make asynchronous`),
+        click: () => this._toggleAsync(eventsFunction),
+      },
+      {
+        label: i18n._(t`Delete`),
+        click: () => this.delete(),
+      },
+      {
+        type: 'separator',
+      },
+      {
+        label: i18n._(t`Copy`),
+        click: () => this._copyEventsFunction(eventsFunction),
+      },
+      {
+        label: i18n._(t`Cut`),
+        click: () => this._cutEventsFunction(eventsFunction),
+      },
+      {
+        label: i18n._(t`Paste`),
+        enabled: Clipboard.has(EVENTS_FUNCTION_CLIPBOARD_KIND),
+        click: () => this._pasteEventsFunction(),
+      },
+      {
+        label: i18n._(t`Duplicate`),
+        click: () => this._duplicateEventsFunction(eventsFunction),
+      },
+    ];
+  }
+
+  renderLeftComponent(i18n: I18nType): ?React.Node {
+    const privateIcon = this.eventFunction.isPrivate() ? (
+      <Tooltip
+        title={
+          <Trans>This function won't be visible in the events editor.</Trans>
+        }
+      >
+        <VisibilityOff fontSize="small" style={styles.tooltip} />
+      </Tooltip>
+    ) : null;
+    const asyncIcon = this.eventFunction.isAsync() ? (
+      <Tooltip
+        title={
+          <Trans>
+            This function is asynchronous - it will only allow subsequent events
+            to run after calling the action "End asynchronous task" within the
+            function.
+          </Trans>
+        }
+      >
+        <AsyncIcon fontSize="small" style={styles.tooltip} />
+      </Tooltip>
+    ) : null;
+    return privateIcon && asyncIcon ? (
+      <>
+        {privateIcon}
+        {asyncIcon}
+      </>
+    ) : (
+      privateIcon || asyncIcon
```

Comment:
It feels a bit complicated. You can use a list for this:
```
const leftIcons = []
if (...) {
  leftIcons.push(...)
}
if (...) {
  leftIcons.push(...)
}
return leftIcons.length > 0 ? leftIcons : null
```
And make sure you add the key props to the array children.